cpp/src/arrow/util/tracing_internal.h

```diff
@@ -114,23 +116,44 @@ class SpanImpl {
 opentelemetry::trace::StartSpanOptions SpanOptionsWithParent(
     const util::tracing::Span& parent_span);
 
-#define START_SPAN(target_span, ...)                                                \
-  auto opentelemetry_scope##__LINE__ =                                              \
-      ::arrow::internal::tracing::GetTracer()->WithActiveSpan(                      \
-          target_span                                                               \
-              .Set(::arrow::util::tracing::Span::Impl{                              \
-                  ::arrow::internal::tracing::GetTracer()->StartSpan(__VA_ARGS__)}) \
-              .span)
-
-#define START_SPAN_WITH_PARENT(target_span, parent_span, ...)                           \
-  auto opentelemetry_scope##__LINE__ =                                                  \
-      ::arrow::internal::tracing::GetTracer()->WithActiveSpan(                          \
-          target_span                                                                   \
-              .Set(::arrow::util::tracing::Span::Impl{                                  \
-                  ::arrow::internal::tracing::GetTracer()->StartSpan(                   \
-                      __VA_ARGS__,                                                      \
-                      ::arrow::internal::tracing::SpanOptionsWithParent(parent_span))}) \
-              .span)
+#define CONFIGURE_TARGET_SPAN(target_span, ...)                             \
```

Comment:
I don't think we need to rearrange all these macros, right? You can keep the original START_SPAN macros and just have START_COMPUTE_SPAN be START_SPAN + a call to SetAttribute. Spans are mutable, there's no requirement that you fully configure them before using them.

Reply:
Mostly I don't want to have a whole bunch of macros for little things.

Reply:
Or actually, maybe just have START_SPAN and CONFIGURE_COMPUTE_SPAN or something. That way we have composable macros instead of having to define duplicate macros for every scenario.

Reply:
I created those macros just for don't repeat the same code twice, but yes I can delete them and copy those lines

Reply:
I don't follow, these macros have repeated code right now